keps/sig-testing/4224-replace-kubeup-clusters-with-kops/README.md

```diff
@@ -0,0 +1,430 @@
+
+<!--
+If applicable, how will the component be upgraded and downgraded? Make sure
+this is in the test plan.
+
+Consider the following in developing an upgrade/downgrade strategy for this
+enhancement:
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to maintain previous behavior?
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to make use of the enhancement?
+-->
+
+Not applicable
+
+### Version Skew Strategy
+
+<!--
+If applicable, how will the component handle version skew with other
+components? What are the guarantees? Make sure this is in the test plan.
+
+Consider the following in developing a version skew strategy for this
+enhancement:
+- Does this enhancement involve coordinating behavior in the control plane and nodes?
+- How does an n-3 kubelet or kube-proxy without this feature available behave when this feature is used?
+- How does an n-1 kube-controller-manager or kube-scheduler without this feature available behave when this feature is used?
+- Will any other components on the node change? For example, changes to CSI,
+  CRI or CNI may require updating that component before the kubelet.
+-->
+
+Not applicable
+
+## Production Readiness Review Questionnaire
+
+<!--
+
+Production readiness reviews are intended to ensure that features merging into
+Kubernetes are observable, scalable and supportable; can be safely operated in
+production environments, and can be disabled or rolled back in the event they
+cause increased failures in production. See more in the PRR KEP at
+https://git.k8s.io/enhancements/keps/sig-architecture/1194-prod-readiness.
+
+The production readiness review questionnaire must be completed and approved
+for the KEP to move to `implementable` status and be included in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and to reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+-->
+
+Not applicable
+
+### Feature Enablement and Rollback
+
+Not Applicable
+
+
+### Dependencies
+
+<!--
+This section must be completed when targeting beta to a release.
+-->
+
+This KEP depends on the following changes in:
+- k/cloud-provider-gcp
+  - images published for both architectures and promoted properly to registry.k8s.io
+  - auth-provider-gcp to be published properly and available at artifacts.k8s.io
+- k/kops
+  - squash various [bugs](https://github.com/kubernetes/kops/issues/15983) with kops 
+
+## Scalability
+
+kops is already used to run scale tests on AWS. We can use it to replace the kube-up scale tests.
```

Comment:
The test on AWS is still visibly less stable than the GCE one:
https://testgrid.k8s.io/kops-misc#ec2-master-scale-performance
vs
https://testgrid.k8s.io/sig-scalability-gce#gce-master-scale-performance


It's by far not clear to me whether it's not because e.g. some settings that are critical to achieve reasonable performance at scale are e.g. not set correctly in kops.

@shyamjvs - FYI


Note - I'm supported for the effort itself, I'm just saying that it may not be as straightforward as you think...
In general, what I would like to see is a diff for flags [fortunately all our components log them on startup] for the existing and new jobs and show that this diff is zero :)

Reply:
kops does expose most of the settings but they do have values that are different from what is set in kube-up clusters. 

https://github.com/kubernetes/kops/pull/15982 tries to close the apiserver differences

I'll look at the scale tests once I can get the serial, disruptive and alpha tests stabilised.

Reply:
We use visibly more for scalability tests - you may want to look at:
https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/sig-scalability/sig-scalability-presets.yaml

Reply:
this is an example of blocker